runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -0,0 +1,251 @@
+        return true;
+    }
+
+    private boolean checkCoreVersion() {
+        try {
+            Class<?> clazz = getBridgeClass("XWalkCoreVersion");
+            int libVersion = (int) new ReflectField(null, clazz, "LIB_VERSION").get();
+            int minLibVersion = (int) new ReflectField(null, clazz, "MIN_LIB_VERSION").get();
+            Log.d("XWalkActivity", "libVersion:" + libVersion + ", minLib:" + minLibVersion);
+            Log.d("XWalkActivity", "sdkVersion:" + mSdkVersion + ", minSdk:" + mMinSdkVersion);
+            Assert.assertTrue(libVersion >= minLibVersion);
+
+            if (mMinSdkVersion > libVersion) {
+                mCoreStatus = LibraryStatus.OLDER_VERSION;
+            } else if (mSdkVersion < minLibVersion) {
+                mCoreStatus = LibraryStatus.NEWER_VERSION;
+            } else {
+                mCoreStatus = LibraryStatus.MATCHED;
+            }
+        } catch (RuntimeException e) {
+            mCoreStatus = LibraryStatus.NOT_FOUND;
+        }
+
+        return mCoreStatus != LibraryStatus.NOT_FOUND;
+    }
+
+    private boolean checkCoreArchitecture() {
+        try {
+            Class<?> clazz = getBridgeClass("XWalkCoreVersion");
+            String targetArch = (String) new ReflectField(null, clazz, "TARGET_ARCH").get();
+            boolean targetIsIA = (targetArch == "ia32");
+
+            clazz = getBridgeClass("XWalkViewDelegate");
+            boolean hostIsIA = (boolean) new ReflectMethod(null, clazz, "isRunningOnIA").invoke();
+
+            Log.d("XWalkActivity", "targetArch:" + targetArch + ", hostIsIA:" + hostIsIA);
+            if (targetIsIA == hostIsIA) return true;
+        } catch (RuntimeException e) {
+        }
+
+        mCoreStatus = LibraryStatus.NOT_FOUND;
+        return false;
+    }
+
+    private Context createBridgeContext() {
+        Context context = null;
+        try {
+            context = XWalkApplication.getApplication().createPackageContext(XWALK_APK_PACKAGE,
+                    Context.CONTEXT_INCLUDE_CODE | Context.CONTEXT_IGNORE_SECURITY);
+            // TODO: check security
+        } catch (NameNotFoundException e) {
+            Log.d("XWalkActivity", "No xwalk core apk");
+        }
+        return context;
+    }
+
+    public boolean isSharedMode() {
+        return mBridgeContext != null;
+    }
+
+    public Object getBridgeObject(Object object) {
+        try {
+            return new ReflectMethod(null, object, "getBridge").invoke();
+        } catch (RuntimeException e) {
+        }
+        return null;
+    }
+
+    public Object getWrapperObject(Object object) {
+        try {
+            return new ReflectMethod(null, object, "getWrapper").invoke();
+        } catch (RuntimeException e) {
+        }
+        return null;
+    }
+
+    public Class<?> getBridgeClass(String name) {
+        try {
+            return mBridgeLoader.loadClass(BRIDGE_PACKAGE + "." + name);
+        } catch (ClassNotFoundException e) {
```

Comment:
The exception here is ok. Null will returned if the object is not from the reflection layer